File: src/main/java/com/numberbox/members/repository/AuthUserInfoRepository.java
package com.numberbox.members.repository;

import com.numberbox.jwt.repository.RefreshTokenInfoRepository;
import com.numberbox.members.entity.Members;
import com.numberbox.members.entity.MembersRole;
import com.numberbox.security.dto.AuthUserInfo;
import com.numberbox.security.dto.AuthUserRole;
import com.numberbox.security.service.JwtRequestUserDetailService;
import com.numberbox.security.service.LoginRequestUserDetailService;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Repository
public class AuthUserInfoRepository implements LoginRequestUserDetailService, JwtRequestUserDetailService {
    MembersRepository membersRepository;
    RefreshTokenInfoRepository refreshTokenInfoRepository;

    public AuthUserInfoRepository(MembersRepository membersRepository,
                                  RefreshTokenInfoRepository refreshTokenInfoRepository) {
        this.membersRepository = membersRepository;
        this.refreshTokenInfoRepository = refreshTokenInfoRepository;
    }

    @Override
    public AuthUserInfo loadUserByUsername(String username) {
        Members member = membersRepository.findByEmail(username);

        if (member == null) return null;

        List<AuthUserRole> roles = new ArrayList<>();
        for (MembersRole role : member.getRole()) {
            roles.add(new AuthUserRole(role.getRoleName(), role.isEnabled()));
        }
        return new AuthUserInfo(member.getEmail(), member.getUserUniqId(), member.getPassword(), roles);
    }

    @Override
    public UUID loadUserIdByRefreshToken(String token) {
        return refreshTokenInfoRepository.finUserUniqIdByToken(token);
    }
}

File: project/infrastructure/orm-adapter/src/main/kotlin/com/numberbox/infra/orm_adpater/repository/auth/AuthUserInfoRepository.kt
package com.numberbox.infra.orm_adpater.repository.auth

import com.numberbox.infra.orm_adpater.repository.member.MemberRefreshTokenInfoRepository
import com.numberbox.infra.orm_adpater.repository.member.MemberRepository
import com.numberbox.modules.auth.control.dto.AuthUserInfo
import com.numberbox.modules.auth.control.dto.AuthUserRole
import com.numberbox.modules.auth.control.service.JwtRequestUserDetailService
import com.numberbox.modules.auth.control.service.LoginRequestUserDetailService
import org.springframework.stereotype.Repository
import java.util.*

@Repository
class AuthUserInfoRepository(
    private val memberRepository: MemberRepository,
    private val refreshTokenInfoRepository: MemberRefreshTokenInfoRepository,
) : LoginRequestUserDetailService, JwtRequestUserDetailService {

    override fun loadUserByUsername(username: String): AuthUserInfo? {
        val member = memberRepository.findByEmail(username) ?: return null

        val roles = member.role.map { AuthUserRole(it.roleName, it.enabled) }
        return AuthUserInfo(member.email, member.userUniqId, member.password, roles)
    }

    override fun loadUserIdByRefreshToken(token: String): UUID? {
        return refreshTokenInfoRepository.findUserUniqIdByToken(token)
    }
}
